### PR TITLE
7 feature 7 home screen

### PR DESCRIPTION
**Pull request by yeludan**

### CODE REVIEW

**Comment by ggykwong** [comment]
Gwen reviews:
- [x] CreateAccountOrLogin
- [x] EnterAGameScreen
- [x] LoginScreen
- [x] RegisterScreen
- [x] LoginController
- [x] LoginPresenter & LoginInterface

Josh reviews:
- [x] ExitOptionsScreen
- [ ] GameScreen
- [x] StorylineInterface
- [ ] CombatController
- [x] StorylineController
- [ ] CombatPresenter & CombatInterface
- [x] StorylinePresenter & StorylineInterface

**Comment by ggykwong** [comment]
RegisterScreen and LoginScreen have correct usage of LoginController, which correctly uses LoginInteractor, and StorylineController to access StorylineInteractor's soundSwitch method.

CreateAccountOrLogin looks good. StorylineInteractor will need to be constructed before CreateAccountOrLogin is in the main program.

EnterAGameScreen has correct usage of LoginController and StorylineController. LoginController also correctly uses LoginInteractor.

**Comment by ggykwong** [comment]
CreateAccountOrLogin looks good. StorylineInteractor will need to be constructed before CreateAccountOrLogin is in the main program.

**Comment by ggykwong** [comment]
EnterAGameScreen has correct usage of LoginController and StorylineController. LoginController also correctly uses LoginInteractor.

File: src/main/java/presenter/StorylinePresenter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+
+    @Override
+    public void display_event(String narration) {  //display string objects
+        gameView.display_event(narration);
+    }
+
+    @Override
+    public void display_event_choices(String narrationChoices) { //display choices string narration
```
Comment (jshchn4) [inline src/main/java/presenter/StorylinePresenter.java:26]: List<String> narrationChoices

File: src/main/java/controller/StorylineController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    public void loadGame() { //load the saved game
+        storylineInteractor.loadGame();
+    }
+
+    public void soundSwitch() {
+        storylineInteractor.soundSwitch();
+    }
+
+    public void updateEventID(Player player, String narrationChoices) {
```
Comment (jshchn4) [inline src/main/java/controller/StorylineController.java]: int choice

**Comment by ggykwong** [comment]
LoginPresenter complies with LoginInterface and LoginInteractor's createAccount and validateLogin methods.

For ExitOptionsScreen--exitGame brings you to the CreateAccountOrLogin UI. Should the user assume that exiting the game automatically logs them out as well? If so, exitGame may also have to initiate the log out process. If the user isn't automatically logged out, they should be brought to the EnterAGameScreen UI instead.

Otherwise, all of the files I have reviewed look good!

File: src/main/java/presenter/StorylinePresenter.java (hunk 1 above)
Comment (jshchn4) [inline src/main/java/presenter/StorylinePresenter.java]: List<String> narrationChoices

File: src/main/java/controller/StorylineController.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        storylineInteractor.updateEventID(player, narrationChoices);
```
Comment (jshchn4) [inline src/main/java/controller/StorylineController.java]: updateEventID(player, choice)

**Review by jshchn4 — CHANGES_REQUESTED** [review]

**Review by ggykwong — CHANGES_REQUESTED** [review]
LoginPresenter complies with LoginInterface and LoginInteractor's createAccount and validateLogin methods.

For ExitOptionsScreen--exitGame brings you to the CreateAccountOrLogin UI. Should the user assume that exiting the game automatically logs them out as well? If so, exitGame has to initiate the log out process. If the user isn't automatically logged out, they should be brought to the EnterAGameScreen UI instead.

Otherwise, all of the files I have reviewed look good!

File: src/main/java/controller/StorylineController.java (hunk 2 above)
Comment (yeludan) [reply src/main/java/controller/StorylineController.java]: fixed!

**Pull request merged**